packages/dialog/src/index.vue

```diff
@@ -0,0 +1,276 @@
+      default: '',
+    },
+    closeOnClickModal: {
+      type: Boolean,
+      default: true,
+    },
+    closeOnPressEscape: {
+      type: Boolean,
+      default: true,
+    },
+    fullscreen: {
+      type: Boolean,
+      default: false,
+    },
+    lockScroll: {
+      type: Boolean,
+      default: true,
+    },
+    modal: {
+      type: Boolean,
+      default: true,
+    },
+    showClose: {
+      type: Boolean,
+      default: true,
+    },
+    title: {
+      type: String,
+      default: '',
+    },
+    openDelay: {
+      type: Number,
+      validator: val => typeof val === 'number',
+      default: 0,
+    },
+    closeDelay: {
+      type: Number,
+      validator: val => typeof val === 'number',
+      default: 0,
+    },
+    top: {
+      type: String,
+      default: '15vh',
+    },
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+    width: {
+      type: String,
+      default: '50%',
+      validator: (val: string) =>
+        ['px', 'rem', 'em', 'vw', '%'].some(unit => val.endsWith(unit)),
+    },
+  },
+  emits: [
+    OPEN_EVENT,
+    OPENED_EVENT,
+    CLOSE_EVENT,
+    CLOSED_EVENT,
+    UPDATE_MODEL_EVENT,
+  ],
+  setup(props, ctx) {
+    // init here
+    return useDialog(props, ctx)
+  },
+
+  render() {
+    if (this.destroyOnClose && !this.modelValue) {
+      return null
+    }
+    const { $slots } = this
+    const closeBtn = this.showClose
+      ? h(
+        'button',
+        {
+          type: 'button',
+          class: 'el-dialog__headerbtn',
+          ariaLabel: 'close',
+          onClick: this.handleClose,
+        },
+        h('i', { class: 'el-dialog__close el-icon el-icon-close' }),
+      )
+      : null
+    const header = h(
+      'div',
+      {
+        class: 'el-dialog__header',
+      },
+      [
+        $slots.header
+          ? $slots.header()
+          : h('span', { class: 'el-dialog__title' }, this.title),
+        closeBtn,
+      ],
+    )
+
+    const body = h(
+      'div',
+      {
+        class: 'el-dialog__body',
+      },
+      $slots.default?.(),
+    )
+
+    const footer = $slots.footer
+      ? h('div', { class: 'el-dialog__footer' }, $slots.footer())
+      : null
+
+    const dialog = h(
+      'div',
+      {
+        ariaModal: true,
+        ariaLabel: this.title || 'dialog',
+        class: [
+          'el-dialog',
+          {
+            'is-fullscreen': this.fullscreen,
+            'el-dialog__center': this.center,
+          },
+          this.customClass,
+        ],
+        ref: 'dialogRef',
+        role: 'dialog',
+        style: this.style,
+        onClick: (e: MouseEvent) => e.stopPropagation(),
+      },
+      [header, body, footer],
+    )
+
+    const trappedDialog = withDirectives(dialog, [[TrapFocus]])
+    const overlay = withDirectives(
+      h(
+        ElOverlay,
+        {
+          mask: this.modal,
+          onClick: this.onModalClick,
+          zIndex: this.zIndex,
+        },
```

Comment:
First: 
Is the behavior of `Overlay` same as `el-dialog__wrapper`? There seems to be missing `el-dialog__wrapper`. 

Second: 
Is it necessary to confirm judge whether the target of the `click` event is `Overlay` in `onModalClick` or `Overlay`'s `onMaskClick`?

![image](https://user-images.githubusercontent.com/33176053/92469843-1689a100-f208-11ea-83e9-e393ac75fe21.png)

Reply:
Answer to the first question: YES

The second one: I think you were trying to say that should we check if the clicking target is `wrapper` or not, right? If I understand you correctly, this won't be necessary since I have already stopped propagation at line 174.

Reply:
OK, I understand.